### PR TITLE
CI 1.22.x rns

### PR DESCRIPTION
**Pull request by aimurphy**
1.22.x & 1.23.x release notes (combined into one release)

## Open items

* Add delegate version in one fixed issue
* Update date & version for actual prod release
* Missing details for CI-11709

## Previews/change summary

* Changed "Bitrise Integrations" to "Bitrise Workflow Steps/Bitrise Steps" to match terminology from Bitrise website.
* Add FF to CI early access features list: https://6615523b3b856200afdfd090--harness-developer.netlify.app/docs/continuous-integration/ci-supported-platforms#harness-ci-features
* Update Bitrise plugin step topic for empty env var handling: https://6615523b3b856200afdfd090--harness-developer.netlify.app/docs/continuous-integration/use-ci/use-drone-plugins/ci-bitrise-plugin#environment-variables
* Release notes: https://6615523b3b856200afdfd090--harness-developer.netlify.app/release-notes/continuous-integration#april-2024

### CODE REVIEW

File: release-notes/continuous-integration.md [hunk 1]
```diff
@@ -37,17 +37,17 @@ Contact [Harness Support](mailto:support@harness.io) if you have any questions.
...
-<!-- ### Version 1.22.x -->
+### Version 1.22.x
 
-<!-- xx Apr 2024 -->
+<!-- 09 Apr 2024 -->
 
-<!-- #### Early access features -->
+#### Early access features
+
+The feature flag `CI_USE_LESS_STRICT_EVALUATION_FOR_MAP_VARS` can resolve some issues where certain variable expressions aren't correctly resolved. Possible use cases include some variables related to looping strategies that incorrectly resolve to null if included in environment variable expressions or empty output variables populating environment variables in child stages. For more information about CI early access features, go to [What's supported for Harness CI](/docs/continuous-integration/ci-supported-platforms). (CI-11882, CI-11305 ZD-57626)
```
Comment (douglas-j-bothwell) [inline release-notes/continuous-integration.md:46]:
```suggestion
The feature flag `CI_USE_LESS_STRICT_EVALUATION_FOR_MAP_VARS` can resolve some issues where certain variable expressions aren't correctly resolved. (CI-11882, CI-11305 ZD-57626)

Here are some issues where this feature flag might be useful:
- Some variables related to looping strategies incorrectly resolve to null if included in environment variable expressions.
- Empty output variables populate environment variables in child stages. 

For more information about CI early access features, go to [What's supported for Harness CI](/docs/continuous-integration/ci-supported-platforms). 
```

File: release-notes/continuous-integration.md [hunk 2]
```diff
@@ -37,17 +37,17 @@ Contact [Harness Support](mailto:support@harness.io) if you have any questions.
...
-UPDATE THE CI EARLY ACCESS PAGE -->
+#### Fixed issues
 
-<!-- #### Fixed issues -->
-<!-- Builds triggered by Bitbucket Server push events have correct date information in the build history. This issue occurred due to missing date information in the `commits` object returned by the Bitbucket Server API. This change requires Harness Delegate version 24.04.XXXXX or later. For information about features that require a specific delegate version, go to the [Delegate release notes](/release-notes/delegate). (CI-11556, ZD-58798) -->
+Builds triggered by Bitbucket Server push events have correct date information in the build history. This issue occurred due to missing date information in the `commits` object returned by the Bitbucket Server API. This change requires Harness Delegate version 24.04.XXXXX or later. For information about features that require a specific delegate version, go to the [Delegate release notes](/release-notes/delegate). (CI-11556, ZD-58798)
```
Comment (douglas-j-bothwell) [inline release-notes/continuous-integration.md:50]:
```suggestion
Builds triggered by Bitbucket Server push events now have correct date information in the build history. This issue occurred due to missing date information in the `commits` object returned by the Bitbucket Server API. This change requires Harness Delegate version 24.04.XXXXX or later. For information about features that require a specific delegate version, go to the [Delegate release notes](/release-notes/delegate). (CI-11556, ZD-58798)
```

**Review by douglas-j-bothwell — APPROVED** [review]
Approved with optional word-tweaks.

File: release-notes/continuous-integration.md [hunk 3]
```diff
@@ -36,17 +36,27 @@ Contact [Harness Support](mailto:support@harness.io) if you have any questions.
...
+* Builds triggered by Bitbucket Server push events now have correct date information in the build history. This issue occurred due to missing date information in the `commits` object returned by the Bitbucket Server API. This change requires Harness Delegate version 24.04.XXXXX or later. For information about features that require a specific delegate version, go to the [Delegate release notes](/release-notes/delegate). (CI-11556, ZD-58798)
+* Fixed an issue where references to deleted/nonexistent secrets sometimes caused pipelines to timeout at the initialize step without any logs. (CI-11891, ZD-60575)
+* Added a fix to trim long environment variables, such as the commit message and PR title, during pod creation in stage initialization. This is currently an opt-in fix for effected customers. If you are using a Kubernetes cluster build infrastructure and experiencing initialization timeout with an error message like `rpc error: code = ResourceExhausted desc = trying to send message larger than max`, contact [Harness Support](mailto:support@harness.io) to enable this fix in your account. (CI-11709, ZD-59521)
```
Comment (douglas-j-bothwell) [inline release-notes/continuous-integration.md:59]:
```suggestion
* Added a fix to trim long environment variables, such as the commit message and PR title, during pod creation in stage initialization. This is currently an opt-in fix for affected customers. If you're using a Kubernetes cluster build infrastructure and experiencing initialization timeout with an error message like `rpc error: code = ResourceExhausted desc = trying to send message larger than max`, contact [Harness Support](mailto:support@harness.io) to enable this fix in your account. (CI-11709, ZD-59521)
```

**Review by douglas-j-bothwell — APPROVED** [review]

**Review by SushrutHarness — APPROVED** [review]
LGTM

**Pull request merged**